File: app/src/main/java/com/jamieadkins/gwent/card/detail/DetailContract.java
package com.jamieadkins.gwent.card.detail;

import android.content.Context;

import com.jamieadkins.commonutils.mvp2.BaseView;
import com.jamieadkins.gwent.data.card.CardDetails;
import com.jamieadkins.gwent.model.GwentCard;

/**
 * Specifies the contract between the view and the presenter.
 */

public interface DetailContract {
    interface View extends BaseView {

        void setLoadingIndicator(boolean active);

        Context getContext();

        void showCard(GwentCard card);
    }

    interface Presenter {

    }
}

File: app/src/main/java/com/jamieadkins/gwent/card/detail/DetailContract.kt
package com.jamieadkins.gwent.card.detail

import com.jamieadkins.gwent.base.MvpPresenter

interface DetailContract {
    interface View {

        fun showScreen(cardDetailsScreenData: CardDetailsScreenData)

        fun showLoadingIndicator()

        fun hideLoadingIndicator()
    }

    interface Presenter : MvpPresenter {
        fun setCardId(cardId: String)
    }
}
